devicetypes/smartthings/zigbee-ceiling-fan-light.src/led-fan-lightings.groovy

```diff
@@ -0,0 +1,242 @@
+/*
+ *  Copyright 2017 SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ *  ZigBee Ceiling Fan Light
+ *
+ *  Author: SAMSUNG LED
+ *  Date: 2021-06-30
+ */
+
+import physicalgraph.zigbee.zcl.DataType
+import groovy.json.JsonOutput
+
+metadata {
+	definition (name: "LED FAN lightings", namespace: "SAMSUNG LED", author: "SAMSUNG LED", genericHandler: "Zigbee") {
+		capability "Actuator"		
+		capability "Configuration"
+		capability "Health Check"		
+
+		// Samsung LED
+		fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006, 0008, 0300", outClusters: "0019", manufacturer: "Samsung Electronics", model: "SAMSUNG-ITM-Z-003", deviceJoinName: "Samsung Light", mnmn: "Samsung Electronics", vid: "SAMSUNG-ITM-Z-003"
+	}
+
+	// UI tile definitions
+	tiles(scale: 2) {
+		multiAttributeTile(name: "switch", type: "lighting", width: 6, height: 4, canChangeIcon: true) {
+			tileAttribute("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label: '${name}', action: "off", icon: "st.switches.light.on", backgroundColor: "#00A0DC", nextState: "turningOff"
+				attributeState "off", label: '${name}', action: "on", icon: "st.switches.light.off", backgroundColor: "#ffffff", nextState: "turningOn"
+				attributeState "turningOn", label: '${name}', action: "off", icon: "st.switches.light.on", backgroundColor: "#00A0DC", nextState: "turningOff"
+				attributeState "turningOff", label: '${name}', action: "on", icon: "st.switches.light.off", backgroundColor: "#ffffff", nextState: "turningOn"
+			}
+			tileAttribute("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action: "switch level.setLevel"
+			}
+		}
+
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label: "", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+		
+		main(["switch"])
+		details(["switch", "refresh", "switchLevel"])
+	}
+}
+
+private getFAN_CLUSTER_VALUE() { 0x0202 }
+private getFAN_STATUS_VALUE() { 0x0000 }
+private getON_OFF_CLUSTER_VALUE() { 0x0006 }
+
+def parse(String description) {
+	// Parse incoming device messages to generate events
+	log.debug "description is $description"
+	def event = zigbee.getEvent(description)
+	def zigbeeMap = zigbee.parseDescriptionAsMap(description)
+	if (event) {
+		sendEvent(event)
+	} else if (description?.startsWith('read attr -')) {
+		if (zigbeeMap.cluster == FAN_CLUSTER_VALUE &&
```

Comment:
@RaihaPark please double check if this comparison have same types on both sides, because as far as I can remember, `parseDescriptionAsMap(desc).cluster` is a string, not an integer and perhaps you should use `clusterInt`